src/content/docs/apm/agents/java-agent/instrumentation/java-agent-instrument-kafka-message-queues.mdx

```diff
@@ -70,17 +87,37 @@ To enable Kafka event collection:
+
+Kafka Streams metrics unlike JMS metrics are not per record. Instead, a transaction begins when a Kafka consumer polls records and then the resulting data gets processed.
+
+If you do wish to create transactions, you need to enable a `kafka-streams-spans-<instrumentation module version number>` module.
+
+For example:
+
+```
+class_transformer:
+  com.newrelic.instrumentation.kafka-streams-spans-3.2.0:
+    enabled: true
+```
+
+Please refer to the [compatability docs](https://docs.newrelic.com/docs/apm/agents/java-agent/getting-started/compatibility-requirements-java-agent/) to see what instrumentation module you need to enable for your Kafka Streams version.
 
 ## Enable Kafka distributed traces [#collect-kafka-distributed-traces]
 
-The Java agent can also collect [distributed traces](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) from Kafka clients. Enabling traces doesn't affect the agent's normal operations: it will still report metric or event data from Kafka.
+The Java agent can also collect [distributed traces](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) from Kafka clients. Since Kafka Streams runs on top of Kafka Clients, the steps to manage distributed tracing also applies. 
+Enabling traces doesn't affect the agent's normal operations: it will still report metric or event data from Kafka.
 
 Impacts and requirements to consider before enabling:
 
 * **The instrumentation adds a 150 to 200 byte payload to message headers.** If your Kafka messages are very small, traces can add significant processing and storage overhead. This additional payload size could cause Kafka to drop messages if they exceed your Kafka messaging size limit. For this reason, we recommend testing out Kafka distributed traces in a dev environment before enabling them in production.
 * Distributed tracing is only available for Kafka client versions 0.11.0.0 or higher.
 * If you have **not** enabled distributed tracing for your app before, read the [Transition guide](/docs/apm/distributed-tracing/getting-started/transition-guide-distributed-tracing) before enabling.
 * To propagate W3C trace context via Kafka message headers, see the [distributed tracing API usage guide](/docs/agents/java-agent/api-guides/guide-using-java-agent-api#trace-calls) for details on APIs that were released in Java agent 6.4.0. Note that adding additional headers to Kafka messages will further increase the payload size. To see these APIs in action, see [Using Java agent trace APIs with Kafka](https://github.com/newrelic/newrelic-java-examples/tree/main/newrelic-java-agent/distributed-tracing/kafka-examples).
+* If you are using Kafka Streams, you need to enable a spans instrumentation module (refer to the [Kafka Streams transaction section](#collect-kafka-streams-transactions)). Since a transaction is not per record, accepting distributed trace headers is not recomended as it will only work for one record.
```

Comment:
```suggestion
* If you are using Kafka Streams, you need to enable a spans instrumentation module (refer to the [Kafka Streams transaction section](#collect-kafka-streams-transactions)). Since a transaction is not per record, accepting distributed trace headers is not recommended as it will only work for one record.
```

Reply:
Same with this one - is the second sentence missing a word? Is it supposed to say "Since a transaction is not recorded per record,"?

Reply:
It would add more clarity. The idea here with Kafka Streams transactions is it starts when many records are polled and processed.